### DISCUSSION TITLE
Transition to SPA

### DISCUSSION THREAD

**Discussion by MatthiasReumann** [TUM-Dev/gocast#845] [Ideas]
### Description
As a side note on our current discussion of a start page redesign (#597), I think it might be worth discussing if we slowly but surely want to start transitioning to a SPA application and therefore use a frontend framework such as VueJS. 

The reasoning behind this is that as TUM-Live grows in features, _go-templates_ + _Alpine_ become slightly unbearable and "hacky". Furthermore, splitting the current system into a `TUM-Live-Core-Service` and `TUM-Live-UI` could potentially decrease the amount of collateral damage produced for every new feature. In this hypothetical world the Core-Service would be a "simple" REST (+GRPC) service.

In my opinion the start page as well as the changes of the course page (as discussed in #791) could be a good place to start. 

### Migration phase
In the migration phase we would have to run both systems simultaneously and configure traffic properly.

### Frontend Frameworks
As for the choice of a frontend framework I propose the following list (More suggestions are welcome!):
* [VueJS](https://vuejs.org) (Big and extensive)
* [mithril.js](https://mithril.js.org) (Small and Fast)
* [preact](https://github.com/preactjs/preact) (Even Smaller, React-based API, Active Community)

### Further notice
In the end it boils down if we want to split up the UI and the Core-System or not. In my estimate, in the long run this will need to happen anyways, so better deal with it now than to deal with the ever increasing complexity of TUM-Live later on.

**Comment by mono424** [comment]
Definitely worth a look is [svelte.js](https://svelte.dev/)! From a developer who used react then went to vue and ended at svelte, Svelte was the easiest to understand and learn and had the best devloper experience omo! :))
